tests/sentry/tempest/test_tempest.py

```diff
@@ -0,0 +1,113 @@
+    def test_fetch_latest_item_id_error(self, mock_fetch):
+        mock_fetch.return_value = "Invalid credentials"
+
+        fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.message == "Seems like the provided credentials are invalid"
+        assert self.credentials.latest_fetched_item_id is None
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_ip_not_allowlisted(self, mock_fetch):
+        mock_fetch.return_value = "IP address not allow-listed"
+
+        fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.message == "Seems like our IP is not allow-listed"
+        assert self.credentials.latest_fetched_item_id is None
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_unexpected_response(self, mock_fetch):
+        mock_fetch.return_value = "Some other error"
+
+        fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.latest_fetched_item_id is None
+        assert self.credentials.message == ""  # No specific message set for unexpected responses
+        mock_fetch.assert_called_once()
+
+    @patch("sentry.tempest.tasks.fetch_latest_id_from_tempest")
+    def test_fetch_latest_item_id_connection_error(self, mock_fetch):
+        mock_fetch.side_effect = Exception("Connection error")
+
+        with self.assertLogs("sentry.tempest.tasks", level="INFO") as cm:
+            fetch_latest_item_id(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.latest_fetched_item_id is None
+        assert self.credentials.message == ""
+        mock_fetch.assert_called_once()
+        assert "Fetching the latest item id failed." in cm.output[0]
+
+    @patch("sentry.tempest.tasks.fetch_crashes_from_tempest")
+    def test_poll_tempest_crashes_task(self, mock_fetch):
+        mock_fetch.return_value = json.dumps({"latest_id": 20002})
+
+        # Set this value since the test assumes that there is already an ID in the DB
+        self.credentials.latest_fetched_item_id = "42"
+        self.credentials.save()
+
+        poll_tempest_crashes(self.credentials.id)
+
+        self.credentials.refresh_from_db()
+        assert self.credentials.latest_fetched_item_id == "20002"
+        mock_fetch.assert_called_once()
+
+    @patch("sentry.tempest.tasks.fetch_crashes_from_tempest")
+    def test_poll_tempest_crashes_invalid_json(self, mock_fetch):
+        mock_fetch.return_value = "not valid json"
+
+        # Set this value since the test assumes that there is already an ID in the DB
+        self.credentials.latest_fetched_item_id = "42"
+        self.credentials.save()
+
+        with self.assertLogs("sentry.tempest.tasks", level="INFO") as cm:
+            poll_tempest_crashes(self.credentials.id)
```

Comment:
This is not how we should test `fetch_crashes_from_tempest` and what happens if the response is invalid. It's better to call directly `fetch_crashes_from_tempest` and make all the asserts